### DISCUSSION TITLE
ENC0100: Adding lambda requires restarting the application.

### DISCUSSION THREAD

**Discussion by dhairya-bhardwaj-dhani** [dotnet/runtime#75798] [Q&A] [answered]
Why does adding any component in razor with child content non-empty results in  `ENC0100: Adding lambda requires restarting the application.` for `dotnet watch run`?  I couldn't find anything on this matter on internet, please help!

Please let me know if hot reload is working for you after adding the following component in blazor wasm template.

```razor
<Counter>Hello</Counter>
```

**Comment by hez2010** [comment, marked as answer]
It doesn't work in .NET 6, but it works in .NET 7.

**Reply by dhairya-cse** [reply]
@hez2010 thanks for the answer, I tried with dotnet 7 and it worked well. However getting the following error while deleting such elements now:

`ENC0055: Accessing captured variable 'this' that hasn't been accessed before in lambda requires restarting the application.`

do you have any ideas about it?
